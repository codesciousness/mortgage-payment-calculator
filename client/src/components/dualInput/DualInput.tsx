import React from 'react';
import NumberInput from '../numberInput/NumberInput';
import Label from '../label/Label';
import { styles } from '../../styles';

type DualInputProps = {
    name: string;
    dollar: number;
    percent: number;
    width: number;
    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const DualInput = ({ name, dollar, percent, width, onChange }: DualInputProps): JSX.Element => {

    return (
        <div style={styles.input}>
            <Label name={name}/>
            <div style={{display: 'flex'}}>
                <NumberInput name={`${name} Dollar`} value={dollar} sign='dollar' width={width*(2/3)} dual onChange={onChange}/>
                <NumberInput name={`${name} Percent`} value={percent} sign='percent' width={width*(1/3)} dual onChange={onChange}/>
            </div>
        </div>
    );
};

export default DualInput;