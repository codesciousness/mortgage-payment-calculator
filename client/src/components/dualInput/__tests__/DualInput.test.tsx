import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import DualInput from '../DualInput';

const name = 'Label Name';
const dollar = '10000';
const percent = '20';
const width = 300;
const onChange = () => null;

describe('<DualInput />', () => {
    test('DualInput component is defined', () => {
        expect(<DualInput name={name} dollar={dollar} percent={percent} width={width} onChange={onChange}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<DualInput name={name} dollar={dollar} percent={percent} width={width} onChange={onChange}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with props', () => {
        render(<DualInput name={name} dollar={dollar} percent={percent} width={width} onChange={onChange}/>);
        expect(screen.getByText('Label Name')).toBeInTheDocument();
    });
    test('renders two NumberInput components', () => {
        render(<DualInput name={name} dollar={dollar} percent={percent} width={width} onChange={onChange}/>);
        expect(screen.getByDisplayValue('10000')).toBeInTheDocument();
        expect(screen.getByDisplayValue('20')).toBeInTheDocument();
    });
});