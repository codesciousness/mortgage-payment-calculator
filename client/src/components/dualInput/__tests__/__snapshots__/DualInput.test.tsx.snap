// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DualInput /> component matches the snapshot 1`] = `
<div
  className="DualInput"
  style={
    Object {
      "marginBottom": "0.75rem",
      "width": 300,
    }
  }
>
  <label
    className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-22ecsy-MuiFormLabel-root-MuiInputLabel-root"
    id="LabelName"
  >
    Label Name
  </label>
  <div
    className="DualInput__container"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <div
        className="NumberInput"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-pijlbd-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
              <b>
                $
              </b>
            </div>
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="LabelNameDollar"
              name="Label Name Dollar"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="10000"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-hdw1oc"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="NumberInput"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-32x3t1-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="LabelNamePercent"
              name="Label Name Percent"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="20"
            />
            <div
              className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
            >
              <b>
                %
              </b>
            </div>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-hdw1oc"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
